types/src/blob.rs

```diff
@@ -489,7 +489,7 @@ mod tests {
 
     #[test]
     fn reconstruct_not_sequence_start() {
-        let len = rand::random::<usize>() % 1024 * 1024;
+        let len = rand::random::<usize>() % (1024 * 1024) + 1;
```

Comment:
after double checking operator precedence etc.. maybe it could be just:
```suggestion
        let len : usize = rand::random_range(1..=(1024*1024) + 1);
```
Upper and lower limits are immediately visible this way, no need to do the maths.

Reply:
sounds good, but for this we'd need to upgrade rand

Reply:
@fl0rek In your suggestion the `+ 1` should be removed